### ISSUE TITLE
Rule proposal: `prefer-consecutive-conditions`

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2416]
### Description
When comparing a variable code below will be easier to understand.
```js
if(0.1<foo<0.5) {
}
```
But JavasScript and TypeScript can't. The alternative is to write code like:
```js
if(0.1<foo && foo<0.5) {
}
```

### Fail

```js
const flag = foo > 0.1 && foo < 0.5;
const flag = foo > 0.1 && 0.5 > foo;
const flag = 0.1 < foo && 0.5 > foo;

if(foo > 0.1 && foo < 0.5) {}
if(foo > 0.1 && 0.5 > foo) {}
if(0.1 < foo && 0.5 > foo) {}
```

### Pass

```js
const flag = 0.1 < foo && foo < 0.5;
if(0.1 < foo && foo < 0.5){}
```

### Proposed rule name

prefer-consecutive-conditions

**Comment by sindresorhus** [comment]
I'm going to pass on this. In my opinion, this results in less readable code.
